pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -352,6 +425,24 @@ func (pc *PackageControllerClient) GetPackageControllerConfiguration() (result s
 	return result, err
 }
 
+// ReconcileDelete removes resources after a full cluster lifecycle cluster is deleted.
+func (pc *PackageControllerClient) ReconcileDelete(ctx context.Context, logger logr.Logger, client client.Client, cluster *anywherev1.Cluster) error {
+	namespace := "eksa-packages-" + cluster.Name
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	if err := client.Delete(ctx, ns); err != nil {
```

Comment:
I don't think delete is idempotent so this will fail if called twice (it can happen for a reentry reconciliation for a deleted cluster, like if a transient error happens).

I think you can make it idempotent but just ignoring with `apierrors.IsNotFound`

Reply:
Yeah, I have a follow-up already for this, just haven't merged it yet. :)